### PR TITLE
UGRID: fix wrong tag ordering

### PR DESCRIPTION
**Pull request by ChristosT**
In ugrid reader order of reading the surface elements is important. 

1.  Added a test case that exposes this issue.
2.  fixed issue and updated ugrid tests

Note: Another way to visually validate the problem and the solution is to use meshio-convert to convert any ugrid that has both quads and triangles into vtk. Without this patch surface elements are not connected properly. Applying the patch produces the expected result. 

Also, I introduced a function in  `test_ugrid.py` (`first()`) to pick elements by type just for convenience. It can be removed if this is needed.

### CODE REVIEW

**Comment by keurfonluu** [comment]
Run ``make format`` to make the linter happy.

File: test/test_ugrid.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
     [
         helpers.tri_mesh,
         helpers.quad_mesh,
-        helpers.quad_tri_mesh,
+        helpers.tri_quad_mesh,
```
Comment (nschloe) [inline test/test_ugrid.py]: Let's test both tri_quad and quad_tri here.

Comment (ChristosT) [reply test/test_ugrid.py]: ok no problem

Comment (ChristosT) [reply test/test_ugrid.py]: it actually helped to find some issues and improve the code. Thanks for the idea.

**Comment by nschloe** [comment]
LGTM, thanks!

**Pull request merged**